From 3ed2d03446cdf53fd10c8ad056d9f86db9923666 Mon Sep 17 00:00:00 2001
From: Peter Korsgaard <jacmet@sunsite.dk>
Date: Fri, 9 Sep 2011 20:35:49 +0200
Subject: [PATCH] FbString: fix build without iconv

Commit 690d926 (introduced FbTk::BidiString) broke building without
HAVE_ICONV, because of wrong variable name and use of iconv_t type.

Signed-off-by: Peter Korsgaard <jacmet@sunsite.dk>
---
 src/FbTk/FbString.cc |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/src/FbTk/FbString.cc b/src/FbTk/FbString.cc
index fd34693..5295a66 100644
--- a/src/FbTk/FbString.cc
+++ b/src/FbTk/FbString.cc
@@ -150,6 +150,7 @@ enum ConvType { FB2X = 0, X2FB, LOCALE2FB, FB2LOCALE, CONVSIZE };
 #ifdef HAVE_ICONV
 static iconv_t *iconv_convs = 0;
 #else
+typedef int iconv_t;
 static int iconv_convs[CONVSIZE];
 #endif // HAVE_ICONV
 
@@ -292,7 +293,7 @@ string recode(iconv_t cd, const string &in) {
 
     return ret;
 #else
-    return str;
+    return in;
 #endif // HAVE_ICONV
 }
 
-- 
1.7.5.4

